src/middleware.js

```diff
@@ -65,23 +70,47 @@ export default function ({ onUnauthorized, ...options }) {
       mode,
     })
 
+    // Make the request
     return http(url, requestOptions)
       .catch(error => {
         const response = error.response || error.message || 'There was an error.'
-        next({
-          type: errorType,
-          payload: { ...rest, response },
-          error: true,
-        })
+        
+        // Send error action
+        if (errorAction) {
+          next(parseAction({
+            action: errorAction,
+            payload: { ...rest, response },
+            error: true,
+          }))
+        }
+
         if (error.status === 401 && defaultConfigOptions.onUnauthorized) {
           next(defaultConfigOptions.onUnauthorized())
         }
       })
-      .then(response =>
-        next({
-          type: successType,
-          payload: { ...rest, response },
-        })
-      )
+      .then(response => {
+
+        // Send success action
+        if (successAction) {
+          next(parseAction({
+            action: successAction,
+            payload: { ...rest, response },
+          }))
+        }
+        
+      })
   }
 }
+
+// Create an action from an action "definition."
+const parseAction = ({ action, payload={}, error=false }) => {
```

Comment:
I'm not sure that setting the defaults for destructured values like this works. I had some issues with this which I have the more explicit merging of options up above.